libmuscle/python/libmuscle/manager/mmp_server.py

```diff
@@ -173,6 +177,10 @@ def _register_instance(
                     ' same version of libmuscle.']
         port_objs = [decode_port(p) for p in ports]
         instance = Reference(instance_id)
+
+        if self._instance_manager:
+            self._instance_manager._requests_out.put(MonitorRequest(instance_id, hostname, pid))
```

Comment:
This accesses a private variable, which may be allowed by Python but not by me, at least in this context. There should be an `InstanceManager.monitor_process()` function that does this.